mongo/bulk_write.go

```diff
@@ -70,9 +71,6 @@ func (bw *bulkWrite) execute(ctx context.Context) error {
 
 		bwErr.WriteConcernError = batchErr.WriteConcernError
 		bwErr.Labels = append(bwErr.Labels, batchErr.Labels...)
-		for i := range batchErr.WriteErrors {
-			batchErr.WriteErrors[i].Index = batchErr.WriteErrors[i].Index + int(opIndex)
-		}
```

Comment:
Hmm, do you think the logic of `mergeResults` may also be incorrect in an unordered bulk write?